flang/lib/Lower/CallInterface.cpp

```diff
@@ -31,7 +31,12 @@ bool Fortran::lower::CallerInterface::hasAlternateReturns() const {
 std::string Fortran::lower::CallerInterface::getMangledName() const {
   const auto &proc = procRef.proc();
   if (const auto *symbol = proc.GetSymbol())
-    return converter.mangleName(*symbol);
+    // Do NOT mangle names for functions/subroutines with BIND(C) attribute.
+    // These could be potential calls to external RTL (i.e OpenMP), that may not
+    // follow the same mangling semantics.
+    return Fortran::semantics::IsBindCProcedure(*symbol)
+               ? proc.GetName()
```

Comment:
Here, you also have to look for a potential bind name in the interface (see my other comments about moving part of your code from `CalleeInterface::getMangledName`  into a `getBindCName` helper and using it here.